unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreen/Scripts/LoadingScreenTimeoutController.cs

```diff
@@ -0,0 +1,118 @@
+
+        private LoadingScreenController loadingScreenController;
+        private CancellationTokenSource timeoutCTS;
+        internal ILoadingScreenTimeoutView view;
+        private IWorldState worldState;
+        private Vector2Int currentDestination;
+
+        internal bool goHomeRequested;
+
+        public LoadingScreenTimeoutController(ILoadingScreenTimeoutView loadingScreenTimeoutView, IWorldState worldState, LoadingScreenController loadingScreenController)
+        {
+            this.view = loadingScreenTimeoutView;
+            this.loadingScreenController = loadingScreenController;
+
+            view.OnExitButtonClicked += ExitClicked;
+            view.OnJumpHomeButtonClicked += GoBackHomeClicked;
+
+            this.worldState = worldState;
+
+            currentEvaluatedTimeout = Application.platform == RuntimePlatform.WebGLPlayer ? LOAD_SCENE_TIMEOUT : WEBSOCKET_TIMEOUT;
+        }
+
+        private async UniTaskVoid StartTimeoutCounter(CancellationToken ct)
+        {
+            if (!await UniTask.Delay(TimeSpan.FromMilliseconds(currentEvaluatedTimeout), cancellationToken: ct).SuppressCancellationThrow())
+                DoTimeout();
+        }
+
+        private void DoTimeout()
+        {
+            IParcelScene destinationScene = worldState.GetScene(worldState.GetSceneNumberByCoords(currentDestination));
+
+            if (destinationScene != null)
+            {
+                Dictionary<string, string> variables = new Dictionary<string, string>
+                {
+                    { "sceneID", destinationScene.sceneData?.id },
+                    { "contentServer", destinationScene.contentProvider?.baseUrl },
+                    { "contentServerBundlesUrl", destinationScene.contentProvider?.assetBundlesBaseUrl },
+                };
+
+                GenericAnalytics.SendAnalytic("scene_loading_failed", variables);
+            }
+
+            if (goHomeRequested)
+                loadingScreenController.RandomPositionRequested();
+            else
+                view.ShowSceneTimeout();
+
+            goHomeRequested = false;
+        }
+
+        public void StartTimeout(Vector2Int newDestination)
+        {
+            timeoutCTS = timeoutCTS.SafeRestart();
+            currentDestination = newDestination;
+            StartTimeoutCounter(timeoutCTS.Token);
+        }
+
+        public void StopTimeout()
+        {
+            //Once the websocket has connected and the first fadeout has been done, its always LOAD_SCENE_TIMEOUT
+            currentEvaluatedTimeout = LOAD_SCENE_TIMEOUT;
+
+            view.HideSceneTimeout();
+            timeoutCTS.SafeCancelAndDispose();
+        }
+
+        private void ExitClicked()
+        {
+#if UNITY_EDITOR
+
+            // Application.Quit() does not work in the editor so
+            // UnityEditor.EditorApplication.isPlaying need to be set to false to end the game
+            UnityEditor.EditorApplication.isPlaying = false;
```

Comment:
Is it the only place we use it? If not I would encapsulate it somewhere because it looks ugly

Reply:
Done, encapsulated it inside the Utils class